### PR TITLE
Feature/hierarchy builder

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

* Add common events related to the hierarchy build process. 
* Update vendor dependencies
* Update import process sequence diagram for hierarchy builder and search import integration

### How to review

Review changes

### Who can review

Anyone

### CODE REVIEW

File: events/observation_import_complete.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package events
+
+import "github.com/ONSdigital/go-ns/avro"
+
+// ObservationImportCompleteEvent is used to notify that all observations of an instance have been imported.
```
Comment (eldeal) [inline events/observation_import_complete.go]: Doc doesnt match type name

File: events/observation_import_complete.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+var observationImportCompleteSchema = `{
+  "type": "record",
+  "name": "observations-import-complete",
+  "fields": [
+    {"name": "job_id", "type": "string"},
+    {"name": "instance_id", "type": "string"}
+  ]
+}`
+
+// ObservationImportCompleteEvent is the Avro schema used when all observations have been imported.
```
Comment (eldeal) [inline events/observation_import_complete.go]: Doc doesnt match type name

File: events/observation_import_complete.go (hunk 1 above)
Comment (gedge) [inline events/observation_import_complete.go]: `Event` should be deleted

File: events/observation_import_complete.go (hunk 2 above)
Comment (gedge) [inline events/observation_import_complete.go]: s/Event/Schema/

**Review by gedge — APPROVED** [review]
again, small amendments, am liking having this in one place

pre-approving

File: events/observation_import_complete.go (hunk 1 above)
Comment (CarlHembrough) [reply events/observation_import_complete.go]: This event was redundant after the design update. I have removed it.

File: events/observation_import_complete.go (hunk 2 above)
Comment (CarlHembrough) [reply events/observation_import_complete.go]: This event was redundant after the design update. I have removed it.

File: events/observation_import_complete.go (hunk 1 above)
Comment (CarlHembrough) [reply events/observation_import_complete.go]: This event was redundant after the design update. I have removed it.

File: events/observation_import_complete.go (hunk 2 above)
Comment (CarlHembrough) [reply events/observation_import_complete.go]: This event was redundant after the design update. I have removed it.

**Pull request merged**